solvebio/query.py

```diff
@@ -1220,3 +1220,26 @@ def fields(self):
             fields = [f for f in fields if f not in self._exclude_fields]
 
         return fields
+
+    def __len__(self):
+        base_len = super(QueryFile, self).__len__()
+        if base_len == float('inf'):
+            raise TypeError(
+                'Unable to determine the number of records for the '
+                'current file-based query'
+            )
+        return base_len
+
+    def __repr__(self):
```

Comment:
rather than make a new `__repr__` please modify the superclass to be like this:

```
def __repr__(self):
    ...
    try:
        total_count = len(self)
    except TypeError:
        total_count = float('inf')
    
    if total_count == 0:
        return 'Query returned 0 results.'

    ...
```